### PR TITLE
Make Google Pay Merchant ID Explicitly Optional

### PR DESCRIPTION
**Pull request by sarahkoop**
### Summary of changes

 - Update doc string to note that `googleMerchantId` is optional
 - Remove setting merchant ID from demo app

 ### Checklist

 - ~[ ] Added a changelog entry~

- @sarahkoop

### CODE REVIEW

File: Card/build.gradle [hunk 1]
```diff
@@ -41,7 +41,7 @@ dependencies {
     testImplementation 'org.skyscreamer:jsonassert:1.4.0'
     testImplementation project(':TestUtils')
     testImplementation project(':ThreeDSecure')
-    testImplementation "androidx.core:core-ktx:+"
+    testImplementation "androidx.core:core-ktx:1.6.0"
```
Comment (sarahkoop) [inline Card/build.gradle:44]: Updating the `core-ktx` dependencies to specific versions is unrelated to this PR, but using `+` is causing tests to fail, and we should be pinning to a specific version.

**Review by sshropshire — APPROVED** [review]
🌊

File: GooglePay/src/main/java/com/braintreepayments/api/GooglePayRequest.java [hunk 2]
```diff
@@ -172,13 +175,17 @@ public void setAllowedCardNetworks(@NonNull String paymentMethodType, @NonNull J
     }
 
     /**
+     * Optional.
```
Comment (Epreuve) [inline GooglePay/src/main/java/com/braintreepayments/api/GooglePayRequest.java:178]: can we add `@Deprecated` on the method and `@deprecated` on the jdoc here just to enforce they'll be going away in the next major version?

**Review by Epreuve — APPROVED** [review]
Small doc nit, but overall 🍰

File: GooglePay/src/main/java/com/braintreepayments/api/GooglePayRequest.java (hunk 2 above)
Comment (sarahkoop) [reply GooglePay/src/main/java/com/braintreepayments/api/GooglePayRequest.java:178]: Good call - added the deprecation tags and an CHANGELOG entry to make it explicit

**Pull request merged**